### ISSUE TITLE
TSPropertySignature has no computed flag

### ISSUE THREAD

**Issue by azz** [eslint/typescript-eslint-parser#262]
**What version of TypeScript are you using?**

2.3.2

**What version of `typescript-eslint-parser` are you using?**

a294afa8158c9c088521eed72b6745eed302361c

**What code were you trying to parse?**

```
var x: { [y]: any };
```

**What did you expect to happen?**

`TSPropertySignature` to have a `computed: true` property.

**What happened?**

No `computed` property.

In prettier, we currently print this as `var x: { y: any }`, which is incorrect.

prettier/prettier#1480

**Comment by soda0289** [comment]
Typescript currently uses `ComputedPropertyName` node type for name in this case. We should convert it into a more ESTree like node. This will effect Interfaces as well. I will look into a fix.

**Comment by soda0289** [comment]
@azz @JamesHenry 
I have changed all of the node types to more estree like. Can you think of anytihng I might have missed?

```js
interface TSPropertySignature {
  type: 'TSPropertySignature';
  optional: boolean;
  computed: boolean;
  key: Identifier;
  typeAnnotation: [ TypeAnnotation ]
}

interface TSMethodSignature {
  type: 'TSMethodSignature';
  optional: boolean;
  computed: boolean;
  key: Identifier;
  params: [ Expression ]
  typeAnnotation: [ TypeAnnotation ]
  typeParameter: [ TypeParameter ]
}

interface TSIndexSignature {
  type: 'TSIndexSignature';
  index: Identifier;
  typeAnnotation: [ TypeAnnotation ]
}

interface TSConstructSignature {
  type: 'TSConstructSignature';
  params: [ Expression ]
  typeAnnotation: [ TypeAnnotation ]
  typeParameter: [ TypeParameter ]
}
```
